### PR TITLE
Add OpenShift templates

### PR DESCRIPTION
**Pull request by semtexzv**
SPM-13

- Add openshift templates
- Add configuration for ocdeployer to deploy to CI
- Update READMEs

### CODE REVIEW

File: openshift/build/database.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: spm-builder-database
```
Comment (Josca) [inline openshift/build/database.yml]: Let's remove prefix `spm-builder-` and other useless prefixes.

File: openshift/build/webserver.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+apiVersion: v1
```
Comment (Josca) [inline openshift/build/webserver.yml]: This file should be removed.

File: openshift/build/database.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+parameters:
+  - name: SOURCE_REPOSITORY_URL
```
Comment (Josca) [inline openshift/build/database.yml]: NAMESPACE parameter is missing.

File: openshift/build/listener.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: spm-builder-listener
+parameters:
+  - name: SOURCE_REPOSITORY_URL
```
Comment (Josca) [inline openshift/build/listener.yml]: NAMESPACE

File: openshift/build/database.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+          uri: ${SOURCE_REPOSITORY_URL}
+        type: Git
+      strategy:
+        type: Docker
+        dockerStrategy:
+          dockerfilePath: ${DOCKERFILE}
+          forcePull: true
+      triggers:
+        - type: ConfigChange
+  - apiVersion: v1
```
Comment (Josca) [inline openshift/build/database.yml]: Put empty line here, please.

File: database/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,11 @@
+FROM centos/postgresql-10-centos7
```
Comment (Josca) [inline database/Dockerfile]: We should use rhel7 based image. Centos is only for dev.

File: Dockerfile [hunk 7]
```diff
@@ -1,24 +1,25 @@
 FROM centos:8
```
Comment (Josca) [inline Dockerfile]: Let's use rhel image. We will save the time for the future.

File: openshift/deploy/database.yml [hunk 8]
```diff
@@ -0,0 +1,98 @@
+apiVersion: v1
```
Comment (Josca) [inline openshift/deploy/database.yml]: There is a quite different deploy config for vmaas database. E.g. there is used ConfigMap resource. Also you can see how are probes used there. Try to ask @jdobes for more info.

File: openshift/deploy/database.yml [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+    displayName: Image tag
+    value: latest
+  - name: VOLUME_SIZE
+    description: Requested volume size for spm-engine-database-data
+    value: 1Gi
+objects:
+  - apiVersion: v1
+    kind: PersistentVolumeClaim
+    metadata:
+      creationTimestamp: null
```
Comment (Josca) [inline openshift/deploy/database.yml]: Please remove params with default values.

File: openshift/deploy/listener.yml [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+              command: [ ./listener/entrypoint.sh ]
+              resources:
+                limits: { cpu: 500m,  memory: 512Mi }
+                requests: { cpu: 200m, memory: 512Mi }
+              env:
+                - { name: KAFKA_ADDRESS, value: platform:9092 }
+                - { name: KAFKA_GROUP, value: spm }
+                - { name: UPLOAD_TOPIC, value: platform.upload.available }
+                - { name: EVENTS_TOPIC, value: platform.inventory.events }
+          dnsPolicy: ClusterFirst
```
Comment (Josca) [inline openshift/deploy/listener.yml]: readinessProbe and livenessProbe are missing

File: openshift/deploy/webserver.yml [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+              command: [ ./webserver/entrypoint.sh ]
+              resources:
+                limits: { cpu: 500m,  memory: 512Mi }
+                requests: { cpu: 200m, memory: 512Mi }
+              env:
+                - { name: POSTGRESQL_HOST, value: spm-engine-database }
+                - { name: POSTGRESQL_USER, value: admin }
+                - { name: POSTGRESQL_PASSWORD, value: passwd }
+                - { name: POSTGRESQL_DATABASE, value: spm }
+          dnsPolicy: ClusterFirst
```
Comment (Josca) [inline openshift/deploy/webserver.yml]: readinessProbe and livenessProbe are missing

File: openshift/deploy/webserver.yml [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+              name: spm-engine-webserver
+              command: [ ./webserver/entrypoint.sh ]
+              resources:
+                limits: { cpu: 500m,  memory: 512Mi }
+                requests: { cpu: 200m, memory: 512Mi }
+              env:
+                - { name: POSTGRESQL_HOST, value: spm-engine-database }
+                - { name: POSTGRESQL_USER, value: admin }
+                - { name: POSTGRESQL_PASSWORD, value: passwd }
+                - { name: POSTGRESQL_DATABASE, value: spm }
```
Comment (Josca) [inline openshift/deploy/webserver.yml]: Other vars are missing (gin-mode, log-level...)

**Comment by Josca** [comment]
@jiridostal can you check it please? Maybe I don't understand to the ticket SPM-14 correctly, but I think it's better to add these things now, as we will need to add it in the future anyway.

File: openshift/deploy/webserver.yml (hunk 11 above)
Comment (semtexzv) [reply openshift/deploy/webserver.yml]: Fixed

File: openshift/deploy/database.yml (hunk 8 above)
Comment (semtexzv) [reply openshift/deploy/database.yml]: Any reason to do it in a separate resource ? Any information apart from password will be static.

File: docker-compose.yml [hunk 13]
```diff
@@ -1,20 +1,21 @@
 version: '3'
 
 services:
-
+  
   db:
     container_name: db
```
[the remark this answers is not in the document]
Comment (semtexzv) [reply docker-compose.yml]: Yes

File: conf/common.env [hunk 14]
```diff
@@ -1,20 +1,17 @@
...
 LOGGING_LEVEL_APP=INFO
 LOGGING_LEVEL_LIBS=WARNING
 
+LOG_LEVEL=debug
+LOG_STYLE=plain
+
 DB_TYPE=postgres
 DB_HOST=db
-DB_NAME=app
+DB_NAME=spm
```
[the remark this answers is not in the document]
Comment (semtexzv) [reply conf/common.env]: Fixd'

**Pull request merged**